src/resources/events/controllers/EventsController.ts

```diff
@@ -20,20 +20,44 @@ import { ResourcePermissionController } from "../../auth/controllers/ResourcePer
 import { Filter } from "../../../generated/models/Filter.generated";
 import { CreateEventRequest } from "../../../generated/models/CreateEventRequest.generated";
 import { AuthUser } from "../../../generated/models/AuthUser.generated";
-import { Params } from "../../../common/parameters/Params";
+import { EventParams } from "../../../common/parameters/Params";
 import { GetEventsResponse } from "../../../generated/models/GetEventsResponse.generated";
 import { Reference } from "../../../generated/models/Reference.generated";
 import { Event } from "../../../generated/models/Event.generated";
 import { getEditableByFilter } from "../../../utils/MetadataUtil";
+import { FilterFactory } from "../../../common/filter/FilterFactory";
 
 const log: debug.IDebugger = debug("app:events-controller");
 
 @Service()
 export class EventsController implements ResourcePermissionController {
-	constructor(public eventsService: EventsService) {}
+	constructor(
+		public eventsService: EventsService,
+		@Inject("FilterFactory") public filterFactory: FilterFactory,
+	) {}
+
+	async listEvents(res: express.Response, pagination: Pagination, params?: EventParams) {
+		let filter: Filter = this.getEventsFilter(params);
+		let isFreeOfChargeFilter = undefined;
+		if (params?.isFreeOfCharge === true) {
+			isFreeOfChargeFilter = this.filterFactory.createExactMatchFilter(
+				"admission.ticketType",
+				"ticketType.freeOfCharge",
+			);
+		}
```

Comment:
Dieser Art von Condition könnte man auch so formulieren:
`const isFreeOfChargeFilter = params?.isFreeOfCharge ? this.filterFactory.createExactMatchFilter(
				"admission.ticketType",
				"ticketType.freeOfCharge",
			) : undefined`

Ist aber eher eine persönliche Vorliebe